lib/Importer/Caffe2ModelLoader.cpp

```diff
@@ -624,6 +628,18 @@ llvm::Error Caffe2ModelLoader::loadOperator(const caffe2::OperatorDef &op) {
       node = G_.createFullyConnected(opName, in, W, B);
     }
 
+    // If number of original input dims is greater then expand the output dims
+    // back with the same axis.
+    if (originalInputDims.size() > 2) {
```

Comment:
You should check the whether the number of elements is the same?

Reply:
If everything is correct it will be but this check is also done by createReshape

Reply:
And you get a coredump? We want to avoid coredump.

Reply:
Yeah right now we do but I have been pushing for putting all those check in to llvm::Errors as well so they will return back to c2. But sure for now I can add an additional check here

Reply:
Is this condition correct? I think it should be conditioned on axis?

Reply:
Do you have an example where this would not work?

Reply:
It will work in terms of correctness but we don't necessarily need to add Reshape whenever input dim is greater than 2, right?

Reply:
We can do `axis != 1` and this should cover any case where output of conv should not be 2 dims right?